packages/at_auth/lib/src/enroll/at_enrollment_base.dart

```diff
@@ -21,6 +21,29 @@ abstract class AtEnrollmentBase {
   ///
   /// Returns a [Future] containing an [AtEnrollmentResponse] representing the
   /// result of the approval/denial of an enrollment.
+  @Deprecated("Use approve and deny methods")
   Future<AtEnrollmentResponse> manageEnrollmentApproval(
       AtEnrollmentRequest atEnrollmentRequest, AtLookUp atLookUp);
+
+  /// Approves an enrollment request.
+  ///
+  /// The [AtEnrollmentNotificationRequest] parameter represents the enrollment request details
+  /// for approving the enrollment request.
+  /// The [atLookUp] parameter is used to perform lookups during approval management.
+  ///
+  /// Returns a [Future] containing an [AtEnrollmentResponse] representing the
+  /// result of the approval/denial of an enrollment.
+  Future<AtEnrollmentResponse> approve(
+      AtEnrollmentNotificationRequest atEnrollmentNotificationRequest, AtLookUp atLookUp);
```

Comment:
Why does the parameter have a different type than the parameter of the `deny` method?

Reply:
OK, I see.

I'm still not sure all of this is as clear as it could be to a developer. Can you lay out here the sequence of (1) what data does the enrolling app send (2) what data does the approving app receive (3) what actions can the approving app take and (4) what statuses can the enrolling app receive?